lib/slack/real_time/client.rb

```diff
@@ -234,13 +234,24 @@ def dispatch(event)
 
       def run_handlers(type, data)
         handlers = store.class.events[type.to_s]
-        handlers.each do |handler|
-          store.instance_exec(data, self, &handler)
+        case @async_handlers
+        when :all 
+          Async.run { handlers_loop(handlers, data) }
+        when :none 
+          handlers_loop(handlers, data)
+        else
+          raise "Invalid Slack::RealTime::Client config! async_handlers must be :all or :none"
```

Comment:
Let's specialize a `InvalidAsyncHandlersError` like we do for `NoConcurrencyError`, or raise a `RuntimeError` explicitly. I would include the current value, so "Invalid value "xyz" config#async_handlers, must be :all or :none." or something like that.

Add a period at the end since this is a full sentence.